src/app/pages/transaction-request/transaction-request.tsx

```diff
@@ -101,21 +105,14 @@ function TransactionRequestBase(): JSX.Element | null {
         {transactionRequest.txType === 'token_transfer' && <StxTransferDetails />}
         {transactionRequest.txType === 'smart_contract' && <ContractDeployDetails />}
         <Formik
-          initialValues={{ fee: '', feeType: Estimations[Estimations.Middle] }}
+          initialValues={initialValues}
           onSubmit={onSubmit}
           validateOnChange={false}
           validateOnBlur={false}
           validateOnMount={false}
           validationSchema={validationSchema}
         >
-          {() => (
-            <>
-              <FeeForm />
-              <SubmitAction />
-              <EditNonceDrawer />
-              <HighFeeDrawer />
-            </>
-          )}
+          {() => <TxRequestFormInner />}
```

Comment:
I understand this inner component exists to access the Formik context, to set the nonce value in the form, but wonder if there's another way of doing it. We use this pattern on the send form, but it becomes confusing as logic then gets arbitrarily shared shared between the two.

To keep it more focused on the one job, I'd rename to _TxRequestForm_ and pass in the children. An API more like:

```jsx
<TxRequestForm>
  <FeeForm />
  <SubmitAction />
  <EditNonceDrawer />
  <HighFeeDrawer />
</TxRequestForm>
```

Alternate suggestion, it could even be called `<TxRequestFormNonceSetter>`, if that's all it does.

Not necessary for now, but would be great to have a layout component for this file, to further hide away style-related details, leaving this component with the responsibility of housing tx request logic only.

Reply:
I like this idea, thx. Will refactor it.